app/EventHandlers/PayPalRequestLogger.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\App\EventHandlers;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class PayPalRequestLogger implements EventSubscriberInterface {
+
+	private string $logFilePath;
+	private array $notificationRoutes;
+
+	public function __construct( string $logFilePath, array $notificationRoutes ) {
+		$this->logFilePath = $logFilePath;
+		$this->notificationRoutes = $notificationRoutes;
+	}
+
+	public static function getSubscribedEvents(): array {
+		return [
+			KernelEvents::REQUEST => 'logPayPalRequest'
+		];
+	}
+
+	public function logPayPalRequest( RequestEvent $event ) {
+		$request = $event->getRequest();
+
+		if ( !in_array( $request->attributes->get( '_route' ), $this->notificationRoutes ) ) {
+			return;
+		}
+
+		try {
+			$filesystem = new Filesystem();
+			$filesystem->appendToFile( $this->logFilePath, $this->getLogLine( $request ) );
+		}
+		catch ( \Exception $e ) {
+			echo $e->getMessage();
+		}
+	}
+
+	private function getLogLine( Request $request ): string {
+		return implode( ', ', [
```

Comment:
I'd use a comma without the added space, just in case Kai wants to open this as a CSV file in LibreOffice.